spec.md

```diff
@@ -15,4 +14,569 @@ The key words "unspecified", "undefined", and "implementation-defined" are to be
 An implementation is not compliant if it fails to satisfy one or more of the MUST, REQUIRED, or SHALL requirements for the protocols it implements.
 An implementation is compliant if it satisfies all the MUST, REQUIRED, and SHALL requirements for the protocols it implements.
 
+## Terminology
+
+| Term              | Definition                                       |
+|-------------------|--------------------------------------------------|
+| Bucket            | A flat organization of container that store files.                          |
+| COSI  System              | Container Orchestration system/ Container Object Storage Implementation, communicates with Plugins using COSI service RPCs.                                     |
+| SP                | Storage Provider, the vendor of a COSI plugin implementation.                                                          |
+| RPC               | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).                                         |
+| Node              | A host where the user workload will be running, uniquely identifiable from the perspective of a Plugin by a node ID. |
+| Plugin            | Aka “plugin implementation”, a gRPC endpoint that implements the COSI Services.                                        |
+| Plugin Supervisor | Process that governs the lifecycle of a Plugin, MAY be the COSI.                                                        |
```

Comment:
Add:
`Provisioner | a generic term meant to describe the combination of a sidecar and driver. "Provisioning" a bucket can mean creating a new bucket or granting access to an existing bucket.`

Reply:
Done